tests/numerics/no_flux_14.output

```diff
@@ -21,20 +21,20 @@
     47 = 0
     68 66:  -0.4889
     68 70:  -0.9881
-    69 67:  -0.4394
-    69 71:  -0.8417
-    86 84:  -0.2948
+    69 67:  -0.4929
+    69 71:  -1.0000
+    86 84:  -0.4142
```

Comment:
these are substantially different. do you have an idea why? and are they better now than they were before?

Reply:
This is what this PR is supposed to fix: The old implementation simply used normal vectors that were not really normal to the surface of the manifold. I think it is surprising that nobody ever observed any issues. My explanation is that this happened only in 3D and only in part of the domain where the `Manifold::normal_vector` function was based on two almost parallel tangents that gave rise to cancellation. Due to being surrounded by proper normal vectors, this never really allowed a substantial flow (or whatever people compute with it) to develop in user codes that people would have noticed.

While looking at these issues, I observed that we had no proper test for those things which is the reason why I introduced two separate tests for the normal vectors `spherical_manifold_{09,10}` to actually look into the normal vectors and not just the no-normal flux constraints and whether the normals make sense: I had no clue before that they didn't and it was definitely not the first place where I looked. I was lucky that I got a floating point exception because two normals on two neighboring faces that should have been identical turned out to be pointing in opposite (!) directions, exactly giving zero when computing the average that raised the exception, the setting I check now in `spherical_manifold_10` and `no_flux_15`.

Reply:
Thanks for the explanation. I'm curious now to see what happens in ASPECT, where we use this function all the time for spherical shell models!

Reply:
Let me know what you observe (I don't follow the aspect development closely nowadays...). I of course hope that this makes things better, so I'd be happy to get this confirmed.

Reply:
It's going to be hard to tell -- we have a lot of failures on dev systems at any given time, and only @tjhei 's official tester succeeds with all tests. But that machine runs 8.5, not deal.II dev. So I think we'll only really find out once we switch the tester to 9.0.